spec/support/order_item_with_service_plan_helper.rb

```diff
@@ -8,18 +8,40 @@
     )
   end
 
+  let(:order_item_params) do
+    {
+      :portfolio_item              => portfolio_item,
+      :service_parameters          => service_parameters,
+      :service_plan_ref            => service_plan_ref,
+      :provider_control_parameters => provider_control_parameters,
+      :order_id                    => order.id,
+      :count                       => 1,
+      :context                     => default_request
+    }
+  end
+
+  let(:pre_order_item) do
+    create(:order_item_with_callback, order_item_params.merge(:process_scope => 'before', :process_sequence => 1)).tap do |item|
+      item.update(:state => 'Completed')
```

Comment:
Can't we just create it with the "Completed" state right away, merging that onto the `order_item_params`?

Reply:
We can't. There is a after_create callback in the model to force the state to be `Created`

Reply:
Ah yeah, good point. Do we *need* to use `:order_item_with_callback` instead of just `:order_item` then?

Reply:
agree. maybe we don't need over complicate the test setup. Hui will help refactor this test.